src/api/BaseServiceApi.ts

```diff
@@ -0,0 +1,83 @@
+import { API_PRODUCTS_URL } from '@/constants/envVariables'
+
```

Comment:
class ProductService implements Interface {
   constructor(api: Interface) {
        this.api = api   
}

getProduct {
    return this.api.get('/tetetete')
}
}


const baseService = new BaseService('api url for products', error)
cosnt productService = new ProductService(baseService)


class translationService implements Interface {
   constructor(api: Interface) {
        this.api = api   
}

getTranslation {
    return this.api.get('/tetetete')
}
}

const baseService = new BaseService('api url for translation', error for tranlsations)
cosnt translationsService = new TranslationService(baseService)